### ISSUE TITLE
The training result is poor after using the weak supervision data set

### ISSUE THREAD

**Issue by woqiaow** [kazuto1011/deeplab-pytorch#109]
Hello, if I want to use weakly supervised data to replace the ground truth value to train the segmented network in the experiment, do I need to modify the code after replacing the dataset?
Because I tried to run this code with the pseudo mask image (10582) obtained by weak supervision, but the result was only 0.03, which actually only recognized the background. Do I need to modify the code?
![image](https://user-images.githubusercontent.com/106650655/195979820-01c16f4b-49d3-4c39-9c53-fadfa9254572.png)

**Comment by kazuto1011** [comment]
Have you checked if your GT files have been loaded successfully?
I think you may also need to debug the class-wise progress for training not only the overall loss.

**Comment by woqiaow** [comment]
Is the GT file you mentioned the ground truth image? Yes, successfully loaded. And I tried to train with train (1464) to produce results. The result was 61%, lower than expected. With train_aug (10582), it could only identify the background. The scores of other categories were 0.
Do you have any other suggestions besides debugging?
Thanks for your help!

**Comment by kazuto1011** [comment]
From your results, I guess your model only produces 0 for any images. 
Sorry for the duplicated question. Are your loaded weakly-supervised labels decoded numbers from 0 to 20 successfully? Does it suit your task that the default setting only treats 255 as the label to be ignored in the loss computation? Since I never assumed the weakly-supervised setting to maintain this repository, I can only suggest general things.

**Comment by woqiaow** [comment]
OK, I see. I will go to debug it. Thank you!
